application/core/LimeMailer.php

```diff
@@ -823,9 +823,19 @@ public function getTokenReplacements()
         $aTokenReplacements["GLOBALOPTINURL"] = App()->getController()
             ->createAbsoluteUrl("/optin/participants", array("surveyid" => $this->surveyId, "token" => $token,"langcode" => $language));
         $this->addUrlsPlaceholders("GLOBALOPTINURL");
-        $aTokenReplacements["SURVEYURL"] = $survey->getSurveyUrl($language, ["token" => $token]);
+        $surveyUrlCreator = new \LimeSurvey\Models\Services\SurveyUrl($language, ["token" => $token]);
```

Comment:
You probably want a `use LimeSurvey\Models\Services\SurveyUrl` statement at top of file to make the code a bit easier on the eye.